pkgs/by-name/so/soundalike/package.nix

```
@@ -0,0 +1,61 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitea,
+  chromaprint,
+  makeWrapper,
+}:
+
+buildGoModule rec {
+  pname = "soundalike";
+  version = "0.1.2";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "derat";
+    repo = "soundalike";
+    tag = "v${version}";
+    hash = "sha256-mpYUVTj3Zll6kNuK5Mdzv1R7k5FZy6XFghhzmAPPVM8=";
+  };
+
+  vendorHash = "sha256-7hRezOBcjB2wsx/SwV519wg3Azh+0kHMcAoc9aYPM3A=";
+
+  nativeBuildInputs = [
+    makeWrapper
+    chromaprint
+  ];
+
+  doCheck = true;
+  # need to grab another repo for music test data
+  testDataVersion = "0.0.1";
+  testData = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "derat";
+    repo = "soundalike-testdata";
+    tag = "v${testDataVersion}";
+    hash = "sha256-7JQTnEjoYiiaQlnxsGcfj/9PYDWAkWq4/oxyczjEMo4=";
+  };
+  preCheck = ''
+    # add soundalike to PATH to be available for unit tests
+    export PATH="$GOPATH/bin:$PATH"
+    # need to symlink test data to a relative path
+    # that the unit tests are expecting
+    ln -sfv ${testData} ./testdata
+  '';
+
+  # soundalike depends on fpcalc (chromparint) at runtime, so we
+  # need to use wrapProgram to make it available
+  postInstall = ''
+    wrapProgram $out/bin/soundalike \
+      --prefix PATH : ${lib.makeBinPath [ chromaprint ]}
+  '';
+
```

```suggestion
  nativeInstallCheckInputs = [ versionCheckHook ];
  versionCheckProgramArg = "-version";
  doInstallCheck = true;
```
Now that the version is printed correctly we could also test for it.

---

Updated in latest commit: https://github.com/NixOS/nixpkgs/pull/388332/commits/bd9ebda4bf20d7b53b76e1509aba7ceacfd9988e